src/full/Agda/TypeChecking/Conversion.hs

```diff
@@ -1212,18 +1212,34 @@ leqSort s1 s2 = (catchConstraint (SortCmp CmpLeq s1 s2) :: m () -> m ()) $ do
       -- Likewise for @Prop@
       (Prop a  , Prop b  ) -> leqLevel a b
 
+      -- Likewise for @SSet@
+      (SSet a  , SSet b  ) -> leqLevel a b
+
       -- @Prop l@ is below @Set l@
       (Prop a  , Type b  ) -> leqLevel a b
       (Type a  , Prop b  ) -> no
 
       -- @Setωᵢ@ is above all small sorts (spelling out all cases
       -- for the exhaustiveness checker)
-      (Inf m   , Inf n   ) ->
-        if m <= n || typeInTypeEnabled || omegaInOmegaEnabled then yes else no
-      (Type{}  , Inf _   ) -> yes
-      (Prop{}  , Inf _   ) -> yes
-      (Inf _   , Type{}  ) -> if typeInTypeEnabled then yes else no
-      (Inf _   , Prop{}  ) -> no
+      (Inf f m , Inf f' n) ->
+        if leqFib f f' && (m <= n || typeInTypeEnabled || omegaInOmegaEnabled) then yes else no
+      (Type{}  , Inf f _) -> if f == IsFibrant then yes else no
```

Comment:
Since `Set l` is below `Setω` and `Setω` is below `SSetω`, shouldn't we also have that `Set l` is below `SSetω` to ensure transitivity?

Reply:
Oh, good catch!